src/events/dns.js

```diff
@@ -0,0 +1,358 @@
+const uuid4 = require('uuid4');
+const dns = require('dns');
+const shimmer = require('shimmer');
+const utils = require('../utils.js');
+const tracer = require('../tracer.js');
+const serverlessEvent = require('../proto/event_pb.js');
+const eventInterface = require('../event.js');
+const errorCode = require('../proto/error_code_pb.js');
+const { isBlacklistURL } = require('.././helpers/events');
+
+const URL_BLACKLIST = {
+    'tc.epsagon.com': 'endsWith',
+};
+
+// Resouece records types
+const rrtypesMethods = {
+    resolveAny: 'ANY',
+    resolve6: 'AAAA',
+    resolve4: 'A',
+    resolveCname: 'CNAME',
+    resolveMx: 'MX',
+    resolveNaptr: 'NAPTR',
+    resolveNs: 'NS',
+    resolvePtr: 'PTR',
+    resolveSoa: 'SOA',
+    resolveSrv: 'SRV',
+    resolveTxt: 'TXT',
+};
+
+/**
+ * Initial Dns event.
+ * @param {string} name Operation name
+ * @returns {Object} Object with dnsEvent and event start time.
+ */
+const initialDnsEvent = (name) => {
+    const startTime = Date.now();
+    const resource = new serverlessEvent.Resource([
+        'dns',
+        'dns',
+        name,
+    ]);
+    const dnsEvent = new serverlessEvent.Event([
+        `dns-${uuid4()}`,
+        utils.createTimestampFromTime(startTime),
+        null,
+        'dns',
+        0,
+        errorCode.ErrorCode.OK,
+    ]);
+    dnsEvent.setResource(resource);
+    return { dnsEvent, startTime };
+};
+
+/**
+ * Adding callback data/error to event, and finalize event.
+ * @param {serverlessEvent.Event} dnsEvent Dns event.
+ * @param {number} startTime Event start time.
+ * @param {Error} err Callback error.
+ * @param {string[] | Object[] | Object} metadata Callback metadata.
+ */
+const finalizeEvent = (dnsEvent, startTime, err, metadata) => {
```

Comment:
maybe try catch this function and call `tracer.addException` if an error occurs (so we are more robust and won't disrupt user's callback)